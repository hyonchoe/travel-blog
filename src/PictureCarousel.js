import React from 'react'
import { Carousel } from 'antd'
import './PictureCarousel.css'

const PictureCarousel = (props) => {

    const onChange = (a, b, c) => {
        console.log(a, b, c)
    }

    const divCarouselImages = (props.pics) ? 
        props.pics.map((picUrl) => {
            return ( 
                <div>
                    <img 
                        src={picUrl}
                        alt="image"
                        className="carouselPicture" />
                </div>
            )
        })
        : null

    return (
        <Carousel draggable={true} afterChange={onChange}>
            {divCarouselImages}
        </Carousel>
    )
}

export default PictureCarousel