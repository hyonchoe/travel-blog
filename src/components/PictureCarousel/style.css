.ant-carousel {
    background: #364d79;
}

.ant-carousel .slick-slide {
    text-align: center;
    height: 160px;
    line-height: 160px;
    overflow: hidden;
}
  
.ant-carousel .slick-slide h3 {
    color: #fff;
}

.carouselPicture {
    display: block;
    max-width: 100%;
    max-height: 100%;
    margin: 0 auto;
}