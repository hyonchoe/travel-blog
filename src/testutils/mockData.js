import moment from 'moment'

const mockData = () => {
    //#region Data values
    const tripId = 'dummytripid'
    const otherTripId = 'otherdummytripid'
    const userId = 'dummyuserid'
    const userName = 'dummyusername'
    const userEmail = 'dummyuseremail'
    const title = 'dummytitle'
    const isPublic = true
    const details = 'dummydetails'
    const loc1 = {
        fmtAddr: 'dummyfmtaddr1',
        latLng: [1, 1],
        city: 'dummycity1',
        state: 'dummystate1',
        country: 'dummycountry1',
    }
    const loc2 = {
        fmtAddr: 'dummyfmtaddr2',
        latLng: [2, 2],
        city: 'dummycity2',
        state: 'dummystate2',
        country: 'dummycountry2',
    }
    const locations = [loc1, loc2]
    const image1 = {
        name: 'dummyname1',
        fileUrlName: 'dummyurlname1',
        S3Url: 'dummys3url1'
    }
    const image2 = {
        name: 'dummyname2',
        fileUrlName: 'dummyurlname2',
        S3Url: 'dummys3url2'
    }
    const image3 = {
        name: 'dummyname3',
        fileUrlName: 'dummyurlname3',
        S3Url: 'dummys3url3'
    }
    const images = [image1, image2, image3]
    const startDate = moment('2020-07-01', 'YYYY-MM-DD')
    const endDate = moment('2020-07-04', 'YYYY-MM-DD')
    //#endregion

    const getTrip = () => ({
        _id: tripId,
        userId,
        userName,
        userEmail,
        title,
        startDate,
        endDate,
        public: isPublic,
        details,
        locations,
        images
    })

    const getLastTripLoaded = () => ({
        _id: tripId,
        startDate: startDate,
        endDate: endDate,
    })

    const getUserInfo = () => ({
        userName,
        userEmail,
    })

    const getAuth0UserInfo = () => ({
        name: 'dummyname',
        email: 'dummyemail',
        sub: 'dummysub'
    })

    return {
        tripId, otherTripId, title, startDate, endDate,
        images, locations, loc1,
        getTrip, getLastTripLoaded, getUserInfo,
        getAuth0UserInfo
    }
}

export default mockData