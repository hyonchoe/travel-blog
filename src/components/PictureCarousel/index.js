import React from 'react'
import { Carousel } from 'antd'
import './style.css'

const PictureCarousel = (props) => {

    const onChange = (a, b, c) => {
        console.log(a, b, c)
    }

    const divCarouselImages = (props.images) ? 
        props.images.map((imgInfo) => {
            return ( 
                <div>
                    <img 
                        src={imgInfo.S3Url}
                        alt={imgInfo.name}
                        className="carouselPicture" />
                </div>
            )
        })
        : null

    return (
        <Carousel draggable={true} afterChange={onChange}>
            {divCarouselImages}
        </Carousel>
    )
}

export default PictureCarousel