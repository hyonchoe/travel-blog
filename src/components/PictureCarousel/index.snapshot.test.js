/**
 * Snapshot tests for PictureCarousel component
 * 
 * Run by running 'npm test' in command line.
 */

import React from 'react'
import { shallow } from 'enzyme'
import mockData from '../../testutils/mockData'
import PictureCarousel from './'

describe('PictureCarousel', () => {
    it('matches snapshot for image divs it hosts', () => {
        const wrapper = shallow(
            <PictureCarousel images={mockData().images} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})