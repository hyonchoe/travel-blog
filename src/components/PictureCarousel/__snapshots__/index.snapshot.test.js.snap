// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PictureCarousel matches snapshot for image divs it hosts 1`] = `
<Carousel
  arrows={false}
  dots={true}
  draggable={true}
>
  <div
    key="0"
  >
    <img
      alt="dummyname1"
      className="carouselPicture"
      src="dummys3url1"
    />
  </div>
  <div
    key="1"
  >
    <img
      alt="dummyname2"
      className="carouselPicture"
      src="dummys3url2"
    />
  </div>
  <div
    key="2"
  >
    <img
      alt="dummyname3"
      className="carouselPicture"
      src="dummys3url3"
    />
  </div>
</Carousel>
`;
